src/core/moves/MetropolisHastingsMove.cpp

```diff
@@ -300,11 +381,34 @@ void MetropolisHastingsMove::performMcmcMove( double prHeat, double lHeat, doubl
 
             if ( the_node->isClamped() )
             {
-                ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                try {
+                    ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                }
+                catch (RbException e)
+                {
+                    ln_likelihood_ratio = RbConstants::Double::neginf;
+
+                    if ( e.getExceptionType() != RbException::MATH_ERROR )
+                    {
+                        throw RbException(e);
```

Comment:
This should probably just be `throw;` -- that rethrows the exception that is currently being handled.

Reply:
This is my explanation of `throw;` with no argument.

Reply:
I've changed this to `throw e`. For me that is more readable than `throw`. Should be the same, right? Now changed in commit 95e107f0.

Reply:
It is actually not the same.  `throw e` constructs a new RbException on the stack, whereas `throw` rethrows the one that was caught, without constructing a new one.

Reply:
Hmm, I can change it to simply use throw, although I find that less readable. Does it make a difference?